src/cmap/conn/mod.rs

```diff
@@ -206,10 +216,23 @@ impl Connection {
         command: Command,
         request_id: impl Into<Option<i32>>,
     ) -> Result<CommandResponse> {
+        if self.partial_message_state.needs_response {
+            Message::read_from(&mut self.stream, &mut self.partial_message_state).await?;
+        }
+
+        if self.partial_message_state.bytes_remaining > 0 {
+            let mut bytes = vec![0u8; self.partial_message_state.bytes_remaining];
+            self.stream.read_exact(&mut bytes).await?;
+            self.partial_message_state.bytes_remaining = 0;
+        }
+
         let message = Message::with_command(command, request_id.into());
         message.write_to(&mut self.stream).await?;
 
-        let response_message = Message::read_from(&mut self.stream).await?;
+        self.partial_message_state.needs_response = true;
```

Comment:
What happens if the future gets dropped midway through `write_to`? We'll probably need a way to flush the unsent data before starting to send new stuff, right?

Reply:
If we send an only partially complete message to the server on a connection, then I'm not sure there's much we can do about that. I think the only two options to finish sending the message on the next write and then read and throw away the response or to just close the connection. My instinct is that an unfinished write will be a lot less common than an unfinished read, as there isn't going to be as much of a delay in writing the data as there is for waiting for the response to come back, so closing the connection in this case doesn't seem like quite as big a deal. I think there's also a case to be made that sending an unfinished message when the user stopped awaiting the future is a lot more likely to cause unexpected behavior for the user, since unlike reading the response of an already sent message, completing a sent message could have side effects on the database.

Because of this, I've updated the PR to drop the connection in the case of an unfinished write.